packages/openapi-v3/src/request-body-decorator.ts

```diff
@@ -0,0 +1,148 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  RequestBodyObject,
+  SchemaObject,
+  ReferenceObject,
+} from '@loopback/openapi-v3-types';
+import {MetadataInspector, ParameterDecoratorFactory} from '@loopback/context';
+import {getSchemaForRequestBody} from './';
+import {OAI3Keys} from './keys';
+import * as _ from 'lodash';
+import {inspect} from 'util';
+
+const debug = require('debug')('loopback:openapi3:metadata:requestbody');
+export const REQUEST_BODY_INDEX = 'x-parameter-index';
+
+/**
+ * Describe the request body of a Controller method parameter.
+ *
+ * A typical OpenAPI requestBody spec contains property
+ * `description`, `required`, and `content`:
+ *
+ * ```ts
+ * requestBodySpec: {
+ *   description: 'a user',
+ *   required: true,
+ *   content: {
+ *     'application/json': {...schemaSpec},
+ *     'application/text': {...schemaSpec},
+ *   },
+ * }
+ * ```
+ *
+ * If the `content` object is not provided, this decorator sets it
+ * as `application/json` by default.
+ * If the `schema` object is not provided in a media type, this decorator
+ * generates it for you based on the argument's type. In this case, please
+ * make sure the argument type is a class decorated by @model from `@loopback/repository`
+ *
+ * The simplest usage is:
+ *
+ * ```ts
+ * class MyController {
+ *   @post('/User')
+ *   async create(@requestBody() user: User) {}
+ * }
+ * ```
+ *
+ * or with properties other than `content`
+ *
+ * ```ts
+ * class MyController {
+ *   @post('/User')
+ *   async create(@requestBody({description: 'a user'}) user: User) {}
+ * }
+ * ```
+ *
+ * or to be more complicated, with your customized media type
+ *
+ * ```ts
+ * class MyController {
+ *   @post('/User')
+ *   async create(@requestBody({
+ *     description: 'a user',
+ *     // leave the schema as empty object, the decorator will generate it.
+ *     content: {'application/text': {}}
+ *   }) user: User) {}
+ * }
+ * ```
+ *
+ * @param requestBodySpec The complete requestBody Object or partial of it.
+ * "partial" for allowing no `content` in spec, for example:
+ * ```
+ * @requestBody({description: 'a request body'}) user: User
+ * ```
+ */
+export function requestBody(requestBodySpec?: Partial<RequestBodyObject>) {
+  return function(target: Object, member: string | symbol, index: number) {
+    // Use 'application/json' as default content if `requestBody` is undefined
+    requestBodySpec = requestBodySpec || {content: {}};
+
+    if (_.isEmpty(requestBodySpec.content))
+      requestBodySpec.content = {'application/json': {}};
+
+    // Get the design time method parameter metadata
+    const methodSig = MetadataInspector.getDesignTypeForMethod(target, member);
```

Comment:
Same as in parameter decorator, can we replace this with `getParameterMetadata` or something that works like that?

Reply:
see comment https://github.com/strongloop/loopback-next/pull/1046#discussion_r171049244